azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/VirtualMachineApi.java

```diff
@@ -18,135 +18,111 @@
+    * The List Virtual Machines operation
     */
-   @Named("ShutdownRole")
-   @POST
-   @Path("/roleinstances/{name}/Operations")
-   @Produces(MediaType.APPLICATION_XML)
-   @ResponseParser(ParseRequestIdHeader.class)
-   @Payload(value = "<ShutdownRoleOperation xmlns=\"http://schemas.microsoft.com/windowsazure\">"
-           + "<OperationType>ShutdownRoleOperation</OperationType></ShutdownRoleOperation>")
-   String shutdown(@PathParam("name") String name);
+   @Named("ListVirtualMachines")
+   @GET
+   @Path("/virtualMachines")
+   @SelectJson("value")
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<VirtualMachine> list();
 
    /**
-    * http://msdn.microsoft.com/en-us/library/jj157189
+    * The Delete Virtual Machine operation
     */
-   @Named("StartRole")
-   @POST
-   @Path("/roleinstances/{name}/Operations")
-   @Produces(MediaType.APPLICATION_XML)
-   @ResponseParser(ParseRequestIdHeader.class)
-   @Payload(value = "<StartRoleOperation xmlns=\"http://schemas.microsoft.com/windowsazure\">"
-           + "<OperationType>StartRoleOperation</OperationType></StartRoleOperation>")
-   String start(@PathParam("name") String name);
+   @Named("DeleteVirtualMachine")
+   @DELETE
+   @Path("/virtualMachines/{name}")
+   @Fallback(Fallbacks.VoidOnNotFoundOr404.class)
+   void delete(@PathParam("name") String name);
 
    /**
-    * https://msdn.microsoft.com/en-us/library/azure/jj157193.aspx
+    * The Restart Virtual Machine operation
     */
-   @Named("GetRole")
-   @GET
-   @Path("/roles/{roleName}")
-   @Produces(MediaType.APPLICATION_XML)
-   @XMLResponseParser(RoleHandler.class)
-   @Fallback(NullOnNotFoundOr404.class)
-   Role getRole(@PathParam("roleName") String roleName);
+   @Named("RestartVirtualMachine")
+   @POST
+   @Path("/virtualMachines/{name}/restart")
+   @Fallback(Fallbacks.VoidOnNotFoundOr404.class)
+   void restart(@PathParam("name") String name);
 
    /**
-    * https://msdn.microsoft.com/library/azure/jj157187.aspx
+    * The start Virtual Machine operation
     */
-   @Named("UpdateRole")
-   @PUT
-   @Path("/roles/{roleName}")
-   @Produces(MediaType.APPLICATION_XML)
-   @ResponseParser(ParseRequestIdHeader.class)
-   String updateRole(@PathParam("roleName") String roleName, @BinderParam(RoleToXML.class) Role role);
+   @Named("StartVirtualMachine")
+   @POST
+   @Path("/virtualMachines/{name}/start")
+   @Fallback(Fallbacks.VoidOnNotFoundOr404.class)
+   void start(@PathParam("name") String name);
 
    /**
-    * The Capture VM Image operation creates a copy of the operating system virtual hard disk (VHD) and all of the data
-    * VHDs that are associated with the Virtual Machine, saves the VHD copies in the same storage location as the original
-    * VHDs, and registers the copies as a VM Image in the image repository that is associated with the specified subscription.
+    * The stop Virtual Machine operation
     */
-   @Named("CaptureVMImage")
+   @Named("StopVirtualMachine")
```

Comment:
Let's document and create a card for features we have no implemented in this api. For example: List all virtual machines in an availability set, list all virtual machine sizes in a region, list all available virtual machine sizes for resizing, save an image that is associated with a vm, stop an deallocate a vm.